### PR TITLE
File copy

### PR DESCRIPTION
**Pull request by itdependsnetworks**
Just a copy of @ktbyers work to upload files, but in a more generic manner. Meaning upload ios images rather than just startup-configurations. 

This port's over functionality from pyntc to napalm.

Related to napalm-automation/napalm-base#93

### CODE REVIEW

File: napalm_ios/ios.py [hunk 1]
```diff
@@ -78,6 +79,23 @@ def __init__(self, hostname, username, password, timeout=60, optional_args=None)
...
+    def file_copy(self, src, dest=None, file_system='flash:'):
+        """
+        SCP file to device filesystem, filesystem defaults to flash:.
+
+        Return None or raise exception
+        """
+        if dest is None:
+            dest = basename(src) 
+        if src:
+            (return_status, msg) = self.scp_file(source_file=src,
```
Comment (dbarrosop) [inline napalm_ios/ios.py:91]: Where is `scp_file` implemented?

Comment (itdependsnetworks) [reply napalm_ios/ios.py:91]: Within ios.py.

Comment (ktbyers) [reply napalm_ios/ios.py:91]: scp_file is in ios.py and largely uses underlying Netmiko file transfer method. 

I would think we would want both a put operation and a get operation.

Comment (ktbyers) [reply napalm_ios/ios.py:91]: If we are going to standardize on file_copy in NAPALM, we can probably just rename scp_file to be file_copy. We can leave a reference to scp_file (pointing to file_copy) so if anyone uses the old name, it doesn't break anything.

I would think you would want both a file put and file get operation (i.e. get file from remote device and put file to remote device).

**Comment by ktbyers** [comment]
@itdependsnetworks I think this is probably going to be resubmitted so I am just going to close it. Just reopen when ready.

**Pull request closed without merging**